### PR TITLE
Include spdlog and cppbenchmark in compression module

### PR DESCRIPTION
**Pull request by StoyanTinchev**
# Description
### Changes that are performed:
* Switched `std::chrono::high_resolution_clock` which was previous measurement of compressor/decompressor classes  -with cppbenchmark.
* Added logs with `spdlog`
* Created singleton class with which you can log easy and efficiently
* Updated decompressor function for creating file which parent directories might not exist + updated corresponding tests to use that function
* Updated `.yml` conan profile to use libstdc++11 which is the new version and supports spdlog.

# Test suite
Benchmarks of the compression/decompression and tests for their major units.

# Environment
OS - macOS Monterey
OS version: 21.1.0
CPU architecture: Apple M1
CPU logical cores: 8
CPU physical cores: 8
RAM total: 8.000 GiB

compiler - GNU gcc-11

--------------------

#### This PR makes changes which ...

- [x] are thoroughly tested
- [x] are well documented

### CODE REVIEW

File: .github/eu_conan_setup.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[settings]
```
Comment (boki1) [inline .github/eu_conan_setup.txt]: Please rename this file to something more informative e.g `ci_conan_profile.yml`.

File: .github/eu_conan_setup.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+os=Linux
+os_build=Linux
+arch=x86_64
+arch_build=x86_64
+compiler=gcc
+compiler.version=11
+compiler.libcxx=libstdc++11
+build_type=Release
+[options]
```
Comment (boki1) [inline .github/eu_conan_setup.txt]: Separate `[]` tags with new lines please.

File: CMakeLists.txt [hunk 3]
```diff
@@ -36,29 +39,31 @@ target_link_libraries(LibEugeneCore
         CONAN_PKG::fmt
         CONAN_PKG::ms-gsl
         CONAN_PKG::outcome
+        CONAN_PKG::catch2
```
Comment (boki1) [inline CMakeLists.txt:42]: Please get it back to normal. This change just expands the diff without any actual value. 😃

File: CMakeLists.txt [hunk 4]
```diff
@@ -36,29 +39,31 @@ target_link_libraries(LibEugeneCore
...
+        target_link_libraries(${component}${test_name} TestMain LibEugeneCore)
+        add_test(NAME ${component}${test_name}
+                COMMAND ${component}${test_name})
         message("   -- Test about '${component}' component added from \"${path_prefix}/${component}.cpp\"")
     endmacro()
 
     message("-- Building core unit tests")
 
     add_library(TestMain src/core/Main.cpp)
-    target_link_libraries(TestMain CONAN_PKG::catch2)
```
Comment (boki1) [inline CMakeLists.txt:54]: I think these changes do not belong in this PR. Please move them to a separate one.

File: src/core/storage/Logger.h [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#ifndef EUGENE_SRC_EUGENE_API_LOGGER_H
+#define EUGENE_SRC_EUGENE_API_LOGGER_H
+
+#include "spdlog/spdlog.h"
+#include <spdlog/sinks/basic_file_sink.h>
+#include <memory>
```
Comment (boki1) [inline src/core/storage/Logger.h]: Please separate different header sources. In my opinion keeping `std` headers at the top, then 3-rd party libs and lastly ours works very well.

File: src/core/storage/btree/Btree.h [hunk 6]
```diff
@@ -57,7 +57,7 @@ class BtreeYAMLPrinter;
 
 template<typename Config = DefaultBtreeConfig>
 requires BtreeConfig<Config>
-class Btree final {
```
Comment (boki1) [inline src/core/storage/btree/Btree.h]: I think there is a slight mistake here 😄. Please fix it.

File: src/core/storage/btree/Tests.cpp [hunk 7]
```diff
@@ -9,20 +9,20 @@ using namespace std::ranges::views;
 
 using namespace internal::btree;
```
Comment (boki1) [inline src/core/storage/btree/Tests.cpp]: Same as with `Btree.h`

File: src/core/storage/btree/BtreePrinter.h [hunk 8]
```diff
@@ -23,7 +23,7 @@ std::string join(InputIt begin, const InputIt end,
 
 template<BtreeConfig Config>
```
Comment (boki1) [inline src/core/storage/btree/BtreePrinter.h]: Same as with `Btree.h`

File: src/core/storage/compression/Benchmarks.cpp [hunk 9]
```diff
@@ -0,0 +1,64 @@
+#include <core/storage/compression/Compressor.h>
+#include <core/storage/compression/Decompressor.h>
+#include <benchmark/cppbenchmark.h>
+
+namespace fs = std::filesystem;
+
+//	67108864 for 1GB txt file
+constexpr int file_size = (32);
```
Comment (boki1) [inline src/core/storage/compression/Benchmarks.cpp]: Why the brackets :smile: ?

File: src/core/storage/Logger.h [hunk 10]
```diff
@@ -0,0 +1,25 @@
+#ifndef EUGENE_SRC_EUGENE_API_LOGGER_H
```
Comment (boki1) [inline src/core/storage/Logger.h]: Use `#pragma once` instead of header guards.

File: src/core/storage/Logger.h [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+#define EUGENE_SRC_EUGENE_API_LOGGER_H
+
```
Comment (boki1) [inline src/core/storage/Logger.h]: I am pretty sure this file is at the wrong place. It is not specific to the `storage` module, so it should be moved to `src/core` instead.

File: src/core/storage/compression/Compressor.h [hunk 12]
```diff
@@ -88,16 +100,16 @@ class CompressorInternal {
...
+			m_all_size += get_file_folder_size(file);
 		}
+
+		Logger::the().log(spdlog::level::info,
+		                  R"(Compressor: started for "{0}" file/files/folder/folders with size: "{1}" bytes)",
+		                  m_files.begin()->c_str(), m_all_size);
+
 		m_total_bits = Folder + 9 * m_files.size();
-		for (const auto &item: m_files) {
+		for (const auto &item : m_files) {
```
Comment (boki1) [inline src/core/storage/compression/Compressor.h:102]: Please do not mix spacing changes with functional ones.

**Review by boki1 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 4 above)
Comment (StoyanTinchev) [reply CMakeLists.txt:54]: Ok, my bad

File: .github/eu_conan_setup.txt (hunk 1 above)
Comment (StoyanTinchev) [reply .github/eu_conan_setup.txt]: Renamed

File: CMakeLists.txt (hunk 3 above)
Comment (StoyanTinchev) [reply CMakeLists.txt:42]: Sure

File: src/core/storage/Logger.h (hunk 5 above)
Comment (StoyanTinchev) [reply src/core/storage/Logger.h]: Exactly here I don't need to separate them but I'll do that refactoring to other files

File: src/core/storage/compression/Benchmarks.cpp (hunk 9 above)
Comment (StoyanTinchev) [reply src/core/storage/compression/Benchmarks.cpp]: I don't understand what you mean

Comment (boki1) [reply src/core/storage/compression/Benchmarks.cpp]: Why `= (32)` and not `= 32` or `(32)` :smile: It is not a big deal.

File: src/core/storage/Logger.h (hunk 5 above)
Comment (boki1) [reply src/core/storage/Logger.h]: :+1:

**Review by boki1 — APPROVED** [review]
Seems good to me. GJ
